core/utils/extended_priority_queue_test.cc

```diff
@@ -0,0 +1,49 @@
+#include "extended_priority_queue.h"
+
+#include <gtest/gtest.h>
+
+#include "random.h"
+
+using bess::utils::extended_priority_queue;
+
+namespace {
+
+// Tests decrease_key_top()
+TEST(ExtendedPriorityQueueTest, DecreseKeyTop) {
+  extended_priority_queue<int> queue;
+  queue.push(1);
+  queue.push(10000);
+  queue.push(100);
+  queue.push(1000);
+  EXPECT_TRUE(std::is_heap(queue.container().begin(), queue.container().end()));
+
+  int& top = queue.mutable_top();
```

Comment:
Maybe add an expect here for what top should be right when you pull it out of the heap?